### PR TITLE
k8s: implement Apply

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

File: internal/k8s/apply.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+func Apply(ctx context.Context, rawYAML string) error {
+	// TODO(dmiller) validate that the string is YAML and give a good error
+	dir, err := ioutil.TempDir("", "tilt")
+	if err != nil {
+		log.Fatal(err)
```
Comment (nicks) [inline internal/k8s/apply.go]: return the error?

File: internal/k8s/apply.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	}
+	defer os.RemoveAll(dir)
+
+	tmpf := filepath.Join(dir, "tilt.yaml")
+	fmt.Println(tmpf)
```
Comment (nicks) [inline internal/k8s/apply.go]: :eyes:

Comment (jazzdan) [inline internal/k8s/apply.go]: whoops will remove

File: internal/k8s/apply.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	contents := []byte(rawYAML)
+	if err := ioutil.WriteFile(tmpf, contents, 0666); err != nil {
+		log.Fatal(err)
```
Comment (nicks) [inline internal/k8s/apply.go]: return err?

**Comment by jazzdan** [comment]
btw, I tested this with the following script:

```go
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/windmilleng/tilt/internal/k8s"
)

func main() {
	yaml := `apiVersion: v1
kind: Service
metadata:
  name: hello-kubernetes
spec:
  type: LoadBalancer
  ports:
  - port: 80
    targetPort: 8080
  selector:
    app: hello-kubernetes
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: hello-kubernetes
spec:
  replicas: 3
  template:
    metadata:
      labels:
        app: hello-kubernetes
    spec:
      containers:
      - name: hello-kubernetes
        image: paulbouwer/hello-kubernetes:1.4
        ports:
        - containerPort: 8080`

	err := k8s.Apply(context.Background(), yaml)
	if err != nil {
		fmt.Println(err.Error())
		os.Exit(1)
	} else {
		fmt.Println("It worked!")
	}
}
```

**Pull request merged**